### PR TITLE
Update kv-store.md

### PR DESCRIPTION
**Pull request by letzya**
### For internal users - please add a Jira DX PR ticket to the subject!

https://tyktech.atlassian.net/browse/DX-588

---
### Preview Link

https://deploy-preview-3050--tyk-docs.netlify.app/docs/nightly/tyk-configuration-reference/kv-store/

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

### CODE REVIEW

File: tyk-docs/content/tyk-configuration-reference/kv-store.md [hunk 1]
```diff
@@ -58,38 +59,87 @@ Example configuration inside `tyk.conf`
...
+For use inside the Tyk configuration file, target URL and listen path, pls use the following notation:
 
 | Store                           | Example|
 | --------------------------------| -----:|
 | Consul                          | `consul://path/to/value`                           |
 | Vault                           | `vault://engine/path/to/secret.actual_secret_name` |
 | Configuration file              | `secrets://value`                                  |
 
-For body transforms and URL rewrites, the prefixes are `$secret_vault.`, `$secret_consul.` and `$secret_conf.`
+#### Note about Valut
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-configuration-reference/kv-store.md]:
```suggestion
#### Note about Vault
```

File: tyk-docs/content/tyk-configuration-reference/kv-store.md [hunk 2]
```diff
@@ -58,38 +59,87 @@ Example configuration inside `tyk.conf`
...
+```curl
+curl \
+  --header "X-Vault-Token: <your_vault_token>" \
+  --request GET \
+  https://vault-server.example.com/v1/secret/tyk?lease=true
+```
 
-### Tyk environment variable usage
-For use inside environment variables, the following secrets are supported:
+```json
```
Comment (letzya) [inline tyk-docs/content/tyk-configuration-reference/kv-store.md]:
```suggestion
```yaml
```

File: tyk-docs/content/tyk-configuration-reference/kv-store.md [hunk 3]
```diff
@@ -58,38 +59,87 @@ Example configuration inside `tyk.conf`
...
 | Consul                          | `consul://path/to/value`                           |
 | Vault                           | `vault://engine/path/to/secret.actual_secret_name` |
 | Configuration file              | `secrets://value`                                  |
 
-For body transforms and URL rewrites, the prefixes are `$secret_vault.`, `$secret_consul.` and `$secret_conf.`
+#### Note about Vault
+In traditional systems, secrets are typically stored individually, each with its own unique key. However, in Vault, it allows for a more flexible approach where multiple secrets can be grouped together and stored under a single key. This grouping allows for better organization and management of related secrets, making it easier to retrieve and manage them collectively. This means that for Vault you use the dot notation to retrieve the exact one we need such as below :
+Example for a secret `gw` under `tyk`:
+
+If one enables the `kv` secrets engine under the path `secret` within Vault using:
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-configuration-reference/kv-store.md]:
```suggestion
1. Enable the `kv` secrets engine under the path `secret` within Vault using:  
```

File: tyk-docs/content/tyk-configuration-reference/kv-store.md [hunk 4]
```diff
@@ -58,38 +59,87 @@ Example configuration inside `tyk.conf`
...
 | Configuration file              | `secrets://value`                                  |
 
-For body transforms and URL rewrites, the prefixes are `$secret_vault.`, `$secret_consul.` and `$secret_conf.`
+#### Note about Vault
+In traditional systems, secrets are typically stored individually, each with its own unique key. However, in Vault, it allows for a more flexible approach where multiple secrets can be grouped together and stored under a single key. This grouping allows for better organization and management of related secrets, making it easier to retrieve and manage them collectively. This means that for Vault you use the dot notation to retrieve the exact one we need such as below :
+Example for a secret `gw` under `tyk`:
+
+If one enables the `kv` secrets engine under the path `secret` within Vault using:  
+   `vault secrets enable -version=2 -path=secret kv`  
+And an arbitrary secret `tyk` with the key `gw` and value `123` is created in Vault:
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-configuration-reference/kv-store.md]:
```suggestion
2. Create an arbitrary secret `tyk` with the key `gw` and value `123` in Vault:  
```

File: tyk-docs/content/tyk-configuration-reference/kv-store.md [hunk 5]
```diff
@@ -58,38 +59,87 @@ Example configuration inside `tyk.conf`
...
-For body transforms and URL rewrites, the prefixes are `$secret_vault.`, `$secret_consul.` and `$secret_conf.`
+#### Note about Vault
+In traditional systems, secrets are typically stored individually, each with its own unique key. However, in Vault, it allows for a more flexible approach where multiple secrets can be grouped together and stored under a single key. This grouping allows for better organization and management of related secrets, making it easier to retrieve and manage them collectively. This means that for Vault you use the dot notation to retrieve the exact one we need such as below :
+Example for a secret `gw` under `tyk`:
+
+If one enables the `kv` secrets engine under the path `secret` within Vault using:  
+   `vault secrets enable -version=2 -path=secret kv`  
+And an arbitrary secret `tyk` with the key `gw` and value `123` is created in Vault:  
+   `vault kv put secret/tyk gw=123`  
+Then to retrieve the secret from within Tyk Gateway, we reference the secret using:
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-configuration-reference/kv-store.md]:
```suggestion
3. To retrieve the secret from within Tyk Gateway, we reference the secret using: 
```

File: tyk-docs/content/tyk-configuration-reference/kv-store.md [hunk 6]
```diff
@@ -58,38 +59,87 @@ Example configuration inside `tyk.conf`
...
+Example for a secret `gw` under `tyk`:
+
+If one enables the `kv` secrets engine under the path `secret` within Vault using:  
+   `vault secrets enable -version=2 -path=secret kv`  
+And an arbitrary secret `tyk` with the key `gw` and value `123` is created in Vault:  
+   `vault kv put secret/tyk gw=123`  
+Then to retrieve the secret from within Tyk Gateway, we reference the secret using:  
+   `TYK_GW_SECRET=vault://secret/tyk.gw`
+
+Retrieving from Vault:
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-configuration-reference/kv-store.md]:
```suggestion
To retrieve the secret from within Vault:
```

File: tyk-docs/content/tyk-configuration-reference/kv-store.md [hunk 7]
```diff
@@ -99,48 +149,42 @@ TYK_GW_USEDBAPPCONFIGS
...
-one key.
+### 4. Body Transforms and URL Rewrites
 
-So we use the dot notation for retrieving the exact one we need such as
-below:
-{{< /note >}}
+For body transforms and URL rewrites, the prefixes are `$secret_vault.`, `$secret_consul.` and `$secret_conf.`
 
 
-If you want to set local "secrets" section as environment variable, you should use the following notation:
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-configuration-reference/kv-store.md:145]:
```suggestion
If you want to set local "secrets" section using environment variables, you should use the following notation:
```

**Review by dcs3spp — COMMENTED** [review]
@letzyaPR content of PR lgtm I have had a read through and added suggestions in comments. Seen there is a hugo build error that is causing the status checks to fail

**Review by joshblakeley — APPROVED** [review]
looks good!

**Comment by dcs3spp** [comment]
@letzya Is this for release 5.1 and release 5?

**Comment by letzya** [comment]
yes please @dcs3spp

**Pull request merged**